File: go/LeetCode/balancedStringSplit.go
package main

func main() {
	
}

func balancedStringSplit(s string) int {
	res :=0
	countL :=0
	countR :=0
	for _,v := range s {
		if v == 'L' {
			countL++
		}else {
			countR++
		}
		if countL==countR {
			 res++
			 countL=0
			 countR=0
		}
	}
	return res
}

File: python3/LeetCode/balancedStringSplit.py
class Solution:
    def balancedStringSplit(self, s:str) -> int:
        res=0
        countL=0
        countR=0
        for i in (s):
            if i == 'L':
                countL+=1
            else:
                countR+=1
            if countR==countL:
                res+=1
                countR=countL=0
        return res